### PR TITLE
Add AutoPlay.

### PR DESCRIPTION
**Pull request by shef3r**
Issue #442

### CODE REVIEW

**Comment by shef3r** [comment]
As mentioned in the issue, with this PR you can run:
`ambie:autoplay` to open the app and immedately start playing whatever was playing before,
and `ambie:autoplay?minimize` to open the app, immedately start playing whatever was playing before, and minimize the app.
I also have an idea to add a `?mini` argument there to open Ambie Mini. What do you think?

**Comment by shef3r** [comment]
I also added an Ambie Mini toggle but for some reason when AMini opens there's no play/pause button? is that intentional?
image

File: src/AmbientSounds.Uwp/App.xaml.cs [hunk 1]
```diff
@@ -7,22 +7,27 @@
 using Microsoft.Toolkit.Uwp.Connectivity;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using System.Xml.Schema;
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/App.xaml.cs:12]: Not needed, please remove

File: src/AmbientSounds.Uwp/App.xaml.cs [hunk 2]
```diff
@@ -7,22 +7,27 @@
...
 using Windows.ApplicationModel;
 using Windows.ApplicationModel.Activation;
 using Windows.ApplicationModel.AppService;
 using Windows.ApplicationModel.Background;
 using Windows.ApplicationModel.Core;
+using Windows.ApplicationModel.DataTransfer;
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/App.xaml.cs:18]: Not needed, please remove

File: src/AmbientSounds.Uwp/App.xaml.cs [hunk 3]
```diff
@@ -7,22 +7,27 @@
...
 using Windows.ApplicationModel.Resources.Core;
 using Windows.Foundation.Collections;
 using Windows.Storage;
+using Windows.System;
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/App.xaml.cs:22]: Not needed, please remove

File: src/AmbientSounds.Uwp/App.xaml.cs [hunk 4]
```diff
@@ -7,22 +7,27 @@
...
 using Windows.System.Profile;
 using Windows.UI;
 using Windows.UI.Notifications;
 using Windows.UI.ViewManagement;
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Navigation;
+using static Humanizer.In;
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/App.xaml.cs:30]: Not needed, please remove

File: src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs [hunk 5]
```diff
@@ -14,42 +18,64 @@ public class ProtocolLaunchController
...
+        if (minimize)
+        {
+            IList<AppDiagnosticInfo> infos = await AppDiagnosticInfo.RequestInfoForAppAsync();
+            IList<AppResourceGroupInfo> resourceInfos = infos[0].GetResourceGroups();
+            await resourceInfos[0].StartSuspendAsync();
+        }
+
+        if (mini)
+        {
+            await _navigator.ToCompactOverlayAsync(CompactViewMode.Focus);
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs:78]: Let's remove this for now. The mini feature is only designed for focusing at the moment. There's no "mini player" that just has a pause/play button anymore. I can add it again in the future, but it's not available right now.

File: src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs [hunk 6]
```diff
@@ -14,42 +18,64 @@ public class ProtocolLaunchController
...
-            _ =_shareService.ProcessShareRequestAsync(shareId);
+            mini = true;
+        }
+        else if (arguments.Contains("minimize"))
+        {
+            minimize = true;
+        }
+        // Both can't really be true at the same time since AMini can't be minimized.
+
+        _player?.Play();
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs:67]: player shouldn't be null here. You should be able to do `_player.Play()`

**Comment by dpaulino** [comment]
> I also added an Ambie Mini toggle but for some reason when AMini opens there's no play/pause button? is that intentional? image

I talked about this in one of the review comments. We should remove the ambie mini-related code since ambie mini doesn't support this scenario right now

**Review by dpaulino — CHANGES_REQUESTED** [review]

File: src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs (hunk 5 above)
Comment (shef3r) [reply src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs:78]: thank youuu, will do

File: src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs (hunk 6 above)
Comment (shef3r) [reply src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs:67]: It crashed on startup for me when I did that, but it works with the question mark, so I'll leave it.

**Review by dpaulino — APPROVED** [review]

**Pull request merged**